libraries/AP_Logger/AP_Logger.h

```diff
@@ -290,6 +288,7 @@ class AP_Logger
     void Write_OABendyRuler(bool active, float target_yaw, bool ignore_chg, float margin, const Location &final_dest, const Location &oa_dest);
     void Write_OADijkstra(uint8_t state, uint8_t error_id, uint8_t curr_point, uint8_t tot_points, const Location &final_dest, const Location &oa_dest);
     void Write_SimpleAvoidance(uint8_t state, const Vector2f& desired_vel, const Vector2f& modified_vel, bool back_up);
+    void Write_Winch(bool healthy, bool thread_end, bool moving, bool clutch, uint8_t mode, float desired_length, float length, float desired_rate, uint16_t tension, float voltage, int8_t temp);
```

Comment:
The reason I left mode as a uint8_t was to ensure AP_Logger and AP_Winch don't both rely on each other.  If this one field becomes an enum defined in AP_Winch then AP_Logger.h will need to include AP_Winch.h and AP_Winch.cpp will then need to include AP_Logger.h.  In the end the logger needs it converted to a simple type to log anyway.

I've just seen other projects get all messed up with a tangle of includes that I prefer not to add them if possible.

Reply:
This probably shows by doing the "easy" method might be better generally - just using `Log_Write(...)` rather than creating all the structures.  Still, the work's done here now.